code/datums/martial.dm

```diff
@@ -51,7 +51,7 @@
 
 	add_logs(A, D, "punched")
 
-	if((D.stat != DEAD) && damage >= 9)
+	if(!D.stat && !D.paralysis && !D.stunned && !D.weakened && damage >= 9)
```

Comment:
Unfortunately we can't use !D.incapacitated() since it also check for restrained() (i.e. handcuffed)